spec/content_spec.rb

```diff
@@ -1,22 +1,33 @@
 # frozen_string_literal: true
 
-require 'content'
+# spec/content_spec.rb
 
-RSpec.describe 'Content' do
-  describe '#initialize' do
-    context 'when input is valid' do
-      input = '"this is the description" #tag1 #tag2 !link'
-      content = Content.new(input)
+RSpec.describe Content do
+  describe "#initialize" do
+    subject(:content) { described_class.new(input) }
 
-      it { expect(content.description).to eq 'this is the description' }
-      it { expect(content.tags).to eq ['#tag1', '#tag2'] }
-      it { expect(content.links).to eq ['!link'] }
+    context "when input is valid" do
+      let(:input) { '"this is the description" #tag1 #tag2 !link' }
+
+      it "parses the description, tags, and links correctly" do
+        aggregate_failures do
+          expect(content.description).to eq("this is the description")
+          expect(content.tags).to eq(["#tag1", "#tag2"])
+          expect(content.links).to eq(["!link"])
+          expect(content.error_message).to be_nil
+        end
+      end
     end
 
-    context 'when input is invalid' do
-      input = '"" #tag1 #tag2 !link'
-      content = Content.new(input)
-      it { expect(content.error_message).to eq 'invalid input' }
+    context "when input is invalid" do
+      let(:input) { '"" #tag1 #tag2 !link' }
+
+      it "sets the error message" do
+        expect(content.error_message).to eq("invalid input")
+        expect(content.description).to be_nil
+        expect(content.tags).to be_nil.or eq([])
+        expect(content.links).to be_nil.or eq([])
```

Comment:
@haroldofurtado actually the tags and links could be nil, the only required field here is the description content, that why even when double quotes are present the input is invalid because no description was provided.